eBCSgen/Parsing/ParseBCSL.py

```diff
@@ -203,6 +204,40 @@ def to_side(self):
     context: CNAME ":" "{" rate_complex ("," rate_complex)* "}"
 """
 
+REGEX_GRAMMAR = r"""
+    !?expression: term ("|" term)*
+
+    ?term: factor+
+
+    ?factor: primary quantifier?
+
+    !quantifier: "??"
+            | "*?"
+            | "+?"
+            | "*+"
+            | "++"
+            | "?+"
```

Comment:
So `regex` package supports all these fancy operators? Have you found any documentation explaining that? We will need to update tutorial and reference them I guess.

Reply:
Yes, I used the information about regular expression syntax from their [docs](https://docs.python.org/3/library/re.html). But some of the extended expressions are not implemented (e.g. `(?...)`, `(?aiLmsux)`, ..., `(?(id/name)yes-pattern|no-pattern)`) as I've never used them and they seemed quite complex to implement. Should I implement them too?

Reply:
okay then, no this is definitely enough, it can be always extended if needed.